clinto/parser.py

```diff
@@ -1,233 +1,47 @@
-                parsers = [v for i, v in chain(six.iteritems(main_module), six.iteritems(vars(module)))
-                           if issubclass(type(v), argparse.ArgumentParser)]
-            if not parsers:
-                f = tempfile.NamedTemporaryFile()
-                try:
-                    ast_source = source_parser.parse_source_file(script_path)
-                    python_code = source_parser.convert_to_python(list(ast_source))
-                    f.write(six.b('\n'.join(python_code)))
-                    f.seek(0)
-                    module = imp.load_source(script_name, f.name)
-                except:
-                    sys.stderr.write('Error while converting {0} to ast:\n'.format(script_path))
-                    self.error = '{0}\n'.format(traceback.format_exc())
-                    sys.stderr.write(self.error)
-                else:
-                    main_module = module.main.__globals__ if hasattr(module, 'main') else globals()
-                    parsers = [v for i, v in chain(six.iteritems(main_module), six.iteritems(vars(module)))
-                           if issubclass(type(v), argparse.ArgumentParser)]
-            if not parsers:
-                sys.stderr.write('Unable to identify ArgParser for {0}:\n'.format(script_path))
-                return
-        self.valid = True
-        parser = parsers[0]
-        self.class_name = script_name
-        self.script_path = script_path
-        self.script_description = getattr(parser, 'description', None)
-        self.script_groups = []
-        self.nodes = OrderedDict()
-        self.script_groups = []
-        non_req = set([i.dest for i in parser._get_optional_actions()])
-        self.optional_nodes = set([])
-        self.containers = OrderedDict()
-        for action in parser._actions:
-            # This is the help message of argparse
-            if action.default == argparse.SUPPRESS:
-                continue
-            node = ArgParseNode(action=action)
-            container = action.container.title
-            container_node = self.containers.get(container, None)
-            if container_node is None:
-                container_node = []
-                self.containers[container] = container_node
-            self.nodes[node.name] = node
-            container_node.append(node.name)
-            if action.dest in non_req:
-                self.optional_nodes.add(node.name)
-
-    def get_script_description(self):
-        return {'name': self.class_name, 'path': self.script_path,
-                'description': self.script_description,
-                'inputs': [{'group': container_name, 'nodes': [self.nodes[node].node_attrs for node in nodes]}
-                           for container_name, nodes in six.iteritems(self.containers)]}
+        return False
 
     @property
-    def json(self):
-        return json.dumps(self.get_script_description())
+    def error(self):
+        if self.parser:
+            return self.parser.error
```

Comment:
I left it as `self.parser.error` since the `error` attribute is set in the init for the base class (it'll always be set). But I did change the default return. Also noticed that `get_script_description` stub was wrongly returning a `None` rather than empty `dict` for the 'json' so I've changed that to `{}`